lib/target/generic_cpp/generic_cxx_codegen.cc

```diff
@@ -313,7 +313,102 @@ std::string GenericCXXCodeGen::EmitType(const halo::Type& type) {
 }
 
 static std::string DeclAsExtern(const std::string& str) {
-  return "extern \"C\" {   " + str + ";\n}\n";
+  return "extern \"C\" {\n" + str + ";\n};\n";
+}
+
+std::string GenericCXXCodeGen::GenerateTestFunc(const Function& func,
+                                                const std::string& func_decl,
+                                                const Instruction& ret_inst) {
+  std::ostringstream oss;
+  oss << "#include \"unittests.h\"\n\n";
+
+  auto convert_data_type = [](DataType dtype) {
+    std::string data_type_str;
+    switch (dtype) {
+      case DataType::FLOAT32:
+        data_type_str = "float";
+        break;
+      case DataType::INT32:
+        data_type_str = "int32_t";
+        break;
+      case DataType::UINT32:
+        data_type_str = "uint32_t";
+        break;
+      case DataType::INT64:
+        data_type_str = "int64_t";
+        break;
+      default:
+        HLCHECK(0);
+    }
+    return data_type_str;
+  };
+
+  if (func.IsEntryFunction()) {
+    if (opts_.dialect == Dialect::CXX_11) {
+      oss << DeclAsExtern(func_decl);
+    }
+    oss << "int main(int argc, char** argv) {\n";
+    oss << "  std::string test_case_dir = argv[argc - 1];\n";
```

Comment:
And a simple check for "argc". For example, if  "argc < 5", prints the usage?

When I first try to run the program manually, I don't know how much arguments are needed. So it just crashed.